feat: add config that causes Serverless to halt if no .env file is found.

neverendingqs opened the pull request:
This change is very similar to https://github.com/colynb/serverless-dotenv-plugin/pull/34, but it was old enough that resolving the merge conflict was difficult, so I went with a new PR. This version of the change is backwards compatible and does not change default behaviour.

This change requires https://github.com/colynb/serverless-dotenv-plugin/pull/66 to work properly. The quality notes below were done with both changes.

Quality notes:
* Functional:
  * Confirmed that if `.env` file is not found and `custom.dotenv` / `custom.dotenv.require` / `custom.dotenv.require.file` is not set, Serverless continues without issues
  * Confirmed that if `.env` file is not found and `custom.dotenv.require.file` is set to `false`, Serverless continues without issues
  * Confirmed that if `.env` file is not found and `custom.dotenv.require.file` is set to `true`, Serverless halts

---

`custom.dotenv.require.file` not set:
```
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: DOTENV: Could not find .env file.
```

`custom.dotenv.require.file` set to `true`:
```
Serverless: Load command upgrade
Serverless: Load command uninstall
 
  Error --------------------------------------------------

  Error: DOTENV: Could not find .env file.
```

neverendingqs committed:
feat: add config that causes Serverless to halt if no .env file is found.

neverendingqs commented on index.js:
```diff
@@ -115,11 +120,19 @@ class ServerlessPlugin {
     } catch (e) {
+      if(this.fileRequired && e.type === errorTypes.FILE_NOT_FOUND) {
+        throw e;
+      }
```
This doesn't look great, but I think is required to:

* Not change the default behaviour
* Not halt on errors EXCEPT when the .env file is not found

neverendingqs commented on README.md:
````diff
@@ -104,6 +104,10 @@ custom:
       - AUTH0_CLIENT_SECRET
 ```
 
+> required: true|false (default false)
````
Let's think of another name to not use up `required` for requiring specific env vars to be set (future feature).

neverendingqs replied:
Brainstorming:

```yaml
required:
  file: true
  variables:
    - API_ENDPOINT
```

to support both file and variable requirements.

neverendingqs commented on index.js:
```diff
@@ -115,11 +120,19 @@ class ServerlessPlugin {
         }
       }
     } catch (e) {
+      if(this.fileRequired && e.type === errorTypes.FILE_NOT_FOUND) {
```
```suggestion
      if(e.type === errorTypes.HALT) {
```
or something similar. No need to be specific, just specify "halt" or "don't halt".

colynb replied:
"halt" is the right word! :-D

neverendingqs committed:
fix: using generic error type instead.

neverendingqs committed:
fix: changing required config to be more extendable in the future.

neverendingqs commented:
Sorry for the delay. I have updated the pull request based on our discussion and updated the testing notes in the pull request description.

This pull request is ready for a re-review when you haev a chance @colynb.
